From f05747c67ed1580a00199f9509cae86b4435a29f Mon Sep 17 00:00:00 2001
From: Michael Lass <lass@mail.uni-paderborn.de>
Date: Mon, 4 Jul 2016 20:00:14 +0200
Subject: [PATCH 2/3] Use ldconfig for boost detection

dpkg is only available on debian based systems. Use ldconfig to determine
the installed version of boost.
---
 pri/boostdetect.pri | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/pri/boostdetect.pri b/pri/boostdetect.pri
index 26367eb..3e44d5f 100644
--- a/pri/boostdetect.pri
+++ b/pri/boostdetect.pri
@@ -29,14 +29,12 @@ for(boost, BOOSTS) {
 contains(LATESTBOOST, 0) {
     unix {
         !macx {
-            BOOSTINFO = $$system(dpkg -s libboost-dev | grep 'Version')
-            BADVERSION = $$find(BOOSTINFO, 1\.54)
+            BOOSTVERSION = $$system(ldconfig -p | grep libboost_filesystem.so | grep -o 1\...\.0 | head -n1)
+            BADVERSION = $$find(BOOSTVERSION, 1\.54)
             !isEmpty(BADVERSION) {
                 message("Boost 1.54 has a bug in a function that Fritzing uses, so download or install some other version")
                 error("Easiest to copy the boost library to .../src/lib/, so that you have .../src/lib/boost_1_xx_0")
-            }
-            isEmpty(BADVERSION) {
-                BOOSTVERSION = $$find(BOOSTINFO, 1\...\.0)
+            } else {
                 !isEmpty(BOOSTVERSION) {
                     LATESTBOOST = installed
                     message("using installed BOOST library")
-- 
2.9.0

